containerized.mk

```diff
@@ -4,6 +4,26 @@ DOCKER_IMAGE_DIR=${GOPATH}/src/${PROJECT_ROOT}
 
 DOCKER_SWARMKIT_DELVE_PORT ?= 2345
 
+# HOW TO USE CONTAINERIZED BUILDS:
+#
+# There are lots of flags that we've added to containerized builds of swarmkit,
+# mostly to support the various swarmkit dev's workflows. These can be kind of
+# confusing, and might feel like they require reading the makefile to fully
+# understand. For clarity, these are the various flags and what they do:
+#
+# DOCKER_SWARMKIT_USE_CONTAINER: Build swarmkit inside of a docker container.
+# DOCKER_SWARMKIT_NOMOUNT: Runs the make target without bindmounting the local
+# 	directory into the container. Useful if you're, say, running tests over a
+# 	network socket. Otherwise, we'll be doing the equivalent of 
+# 	-v ~/go/src/github.com/docker/swarmkit:/go/src/github.com/docker/swarmkit
+# DOCKER_SWARMKIT_USE_DELVE: Use the Delve debugger
+# DOCKER_SWARMKIT_DELVE_PORT: The port for hooking up the Delve debugger
+# DOCKER_SWARMKTI_USE_DOCKER_SYNC: uses the Docker Sync tool, which can speed
+# 	up operations when running on Docker for Windows or Docker for Mac
+# DOCKER_SWARMKIT_DOCKER_RUN_FLAGS: add extra flags to the docker run command
```

Comment:
Nope that one is internal :) shouldn't be set externally. Maybe we should prefix it with a `_` to convey that?